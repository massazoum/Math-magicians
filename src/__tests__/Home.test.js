import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Home from '../components/HomePage';
import renderer from 'react-test-renderer';

describe('Home component', () => {
  test('renders home page content', () => {
    render(<Home />);

    // Assert that the heading and paragraphs are rendered
    expect(screen.queryByText('Welcome to the home page')).toBeTruthy();
    expect(
      screen.getByText(
        'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Recusandae modi ratione in? Nobis explicabo cupiditate doloremque pariatur maxime ad, facere accusantium quam fuga tempore! Sequi, autem minima perferendis nulla labore expedita accusamus molestias?'
      )
    ).toBeTruthy();
    expect(
      screen.getByText(
        'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Recusandae modi ratione in? Nobis explicabo cupiditate doloremque pariatur maxime ad, facere accusantium quam fuga tempore! Sequi.'
      )
    ).toBeTruthy();
    // Assert the styling of the component
    const component = screen.getByTestId('home-component');
    expect(component).toHaveStyle('color: black');
    expect(component).toHaveStyle('font-size: 14px');
    expect(component).toHaveStyle('font-family: poppins, sans-serif');
  });
});

test('Homepage.js component renders correctly', () => {
 const component = renderer.create(<Home />);
 const tree = component.toJSON();
 expect(tree).toMatchSnapshot();
});