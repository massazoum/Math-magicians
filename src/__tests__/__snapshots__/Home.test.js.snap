// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage.js component renders correctly 1`] = `
<div
  data-testid="home-component"
  style={
    Object {
      "color": "black",
      "fontFamily": "poppins, sans-serif",
      "fontSize": "14px",
    }
  }
>
  <h3>
    Welcome to the home page
  </h3>
  <p>
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Recusandae modi ratione in? Nobis explicabo cupiditate doloremque pariatur maxime ad, facere accusantium quam fuga tempore! Sequi, autem minima perferendis nulla labore expedita accusamus molestias?
  </p>
  <p>
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Recusandae modi ratione in? Nobis explicabo cupiditate doloremque pariatur maxime ad, facere accusantium quam fuga tempore! Sequi.
  </p>
</div>
`;
